modules/elasticsearch/main.tf

```
@@ -0,0 +1,110 @@
+locals {
+  enabled = module.this.enabled
+
+  dns_zone_id                   = module.dns_delegated.outputs.default_dns_zone_id
+  vpc_id                        = module.vpc.outputs.vpc_id
+  vpc_default_security_group    = module.vpc.outputs.vpc_default_security_group_id
+  vpc_private_subnet_ids        = module.vpc.outputs.private_subnet_ids
+  elasticsearch_domain_endpoint = format("/%s/%s/%s", "elasticsearch", module.this.name, "elasticsearch_domain_endpoint")
+  elasticsearch_kibana_endpoint = format("/%s/%s/%s", "elasticsearch", module.this.name, "elasticsearch_kibana_endpoint")
+  elasticsearch_admin_password  = format("/%s/%s/%s", "elasticsearch", module.this.name, "password")
+}
+
+locals {
+  create_password        = local.enabled && length(var.elasticsearch_password) == 0
+  elasticsearch_password = local.create_password ? join("", random_password.elasticsearch_password.*.result) : var.elasticsearch_password
+}
+
+module "elasticsearch" {
+  source  = "cloudposse/elasticsearch/aws"
+  version = "0.33.0"
+
+  security_groups                = [local.vpc_default_security_group]
+  vpc_id                         = local.vpc_id
+  subnet_ids                     = local.vpc_private_subnet_ids
+  zone_awareness_enabled         = length(local.vpc_private_subnet_ids) > 1 ? true : false
+  elasticsearch_version          = var.elasticsearch_version
+  instance_type                  = var.instance_type
+  instance_count                 = length(local.vpc_private_subnet_ids)
+  availability_zone_count        = length(local.vpc_private_subnet_ids)
+  encrypt_at_rest_enabled        = true //var.encrypt_at_rest_enabled
```

```suggestion
  encrypt_at_rest_enabled        = var.encrypt_at_rest_enabled
```

`var.encrypt_at_rest_enabled` Is set in the defaults and also is mentioned in the usage documentation. It should not be ignored.